packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -16,25 +16,42 @@ limitations under the License.
 */
 import '@material/mwc-notched-outline';
 
+import {MDCLineRippleFoundation} from '@material/line-ripple/foundation.js';
 import {addHasRemoveClass, classMap, FormElement, html, property, PropertyValues, query, TemplateResult} from '@material/mwc-base/form-element.js';
 import {floatingLabel, FloatingLabel} from '@material/mwc-floating-label';
 import {lineRipple, LineRipple} from '@material/mwc-line-ripple';
 import {NotchedOutline} from '@material/mwc-notched-outline';
 import {MDCTextFieldAdapter, MDCTextFieldInputAdapter, MDCTextFieldLabelAdapter, MDCTextFieldLineRippleAdapter, MDCTextFieldOutlineAdapter, MDCTextFieldRootAdapter} from '@material/textfield/adapter.js';
+import {MDCTextFieldCharacterCounterFoundation} from '@material/textfield/character-counter/foundation.js';
 import MDCTextFieldFoundation from '@material/textfield/foundation.js';
 import {ifDefined} from 'lit-html/directives/if-defined.js';
 
 import {characterCounter, CharacterCounter} from './character-counter/mwc-character-counter-directive.js';
 
+// must be done to get past lit-analyzer checks
+declare global {
+  interface Element {
+    foundation?: MDCLineRippleFoundation|MDCTextFieldCharacterCounterFoundation;
```

Comment:
(not blocking this PR, just discussion) There must be a better way to apply directives to an element that doesn't require inventing a fake property. @justinfagnani thoughts on that?

Reply:
There isn't currently. This is a pretty common need though...

Reply:
Is that why all the `foundation` properties are renamed in this PR?

Reply:
correct!

Reply:
https://github.com/material-components/material-components-web-components/blob/d6f763fb51d20ee669f4aeef850611506ff42b94/packages/textfield/src/mwc-textfield-base.ts#L35